src/compare_main.cpp

```diff
@@ -292,24 +302,16 @@ int pandora_compare(int argc, char *argv[]) {
     // load read index
     std::cout << now() << "Loading read index file " << read_index_fpath << std::endl;
     auto samples = load_read_index(read_index_fpath);
+    std::vector<std::string> sample_names;
+    for (const auto &sample_pair : samples) sample_names.push_back(sample_pair.first);
 
-    auto pangraph = std::make_shared<pangenome::Graph>(pangenome::Graph());
-    auto pangraph_sample = std::make_shared<pangenome::Graph>(pangenome::Graph());
-
-    auto minimizer_hits = std::make_shared<MinimizerHits>(MinimizerHits(100000));
-
-    Fastaq consensus_fq(true, true);
-
-    vector<KmerNodePtr> kmp;
-    vector<LocalNodePtr> lmp;
-    vector<vector<uint32_t>> read_overlap_coordinates;
+    auto pangraph = std::make_shared<pangenome::Graph>(sample_names);
+    std::vector<uint32_t> exp_depth_covgs;
 
     // for each sample, run pandora to get the sample pangraph
-    std::vector<uint32_t> exp_depth_covgs;
-    uint32_t sample_id = 0;
-    for (const auto &sample: samples) {
-        pangraph_sample->clear();
-        minimizer_hits->clear();
+    for (uint32_t sample_id = 0; sample_id < samples.size(); ++sample_id) {
```

Comment:
Why the switch from range-based `for`?

Reply:
We need the `sample_id` inside the loop now